### ISSUE TITLE
Weird bug

### ISSUE THREAD

**Issue by jahodovykolac** [yairm210/Unciv#3739]
On version 3.13.10, I struggle to create a game with Large rectangular map of any type with World Wrap. Every second turn (literally) is just blue, no map, no units, no anything, leaving a popup saying 'Not enough memory to load game!'. When I change the shape to Hexagonal, the game crashes.

**Comment by yairm210** [comment]
I assume that you're on Android?

**Comment by yairm210** [comment]
If so, then it's Exactly What It Says On The Tin. Your device doesn't have enough memory to hold the large maps in memory.

**Comment by jahodovykolac** [comment]
Okay, I will comment here. 
A little oopsie, I found out the older game was Medium, not Large.
And yes, I know the number of objects increases over time, but this is like: Turn 0 - works, Turn 1 - doesn't work, Turn 2 - works, Turn 3 - doesn't work...

**Comment by jahodovykolac** [comment]
I can even select a unit by pressing Next unit, but there are no tiles.

**Comment by yairm210** [comment]
The World Wrap basically creates 'copies' of the tiles on right and left sides, in order to give the illusion that you're scrolling infinitely. The 'alternate turns it works' makes a lot of sense to me - what's happening is that when you click 'next turn' we create a map of the next turn behind the scenes and change it out, so that the user can still interact with the old map while the new one is loading.
This means that during that intermediate time, there are actually 2 maps in parallel, meaning double memory.
But! If the creation fails because there's not enough memory, that means that during that rturn you will actually be left with NO map at all! And then on the 'next turn' when we create a new map, it will be the only map, so no memory problem!

TL;DR your device can probably handle a single map, but not 2 of them, so the 'handover' at the "next turn" causes it to exceed its memory limits.

**Comment by jahodovykolac** [comment]
Okay, is there any solution for that or I will just not play on this type of map?

**Comment by yairm210** [comment]
Not anything specific that I can think of

**Comment by jahodovykolac** [comment]
Maybe if the game cleared the old map instead of the new map?
